src/java/org/apache/cassandra/service/StartupChecks.java

```diff
@@ -285,6 +289,77 @@ public void execute()
         }
     };
 
+    public static final StartupCheck checkReadAheadKbSetting = new StartupCheck()
+    {
+        // This value is in KB.
+        private static final long MAX_RECOMMENDED_READ_AHEAD_KB_SETTING = 128;
+
+        @Override
+        public void execute() throws StartupException
+        {
+            if (!FBUtilities.isLinux)
+                return;
+
+            String[] dataDirectories = DatabaseDescriptor.getRawConfig().data_file_directories;
+            Map<String, String> blockDevices = new HashMap<String, String>();
+            for (String dataDirectory : dataDirectories)
+            {
+                try
+                {
+                    Path p = Path.of(dataDirectory);
+                    FileStore fs = Files.getFileStore(p);
+
+                    String blockDirectory = fs.name();
+                    if(StringUtils.isNotEmpty(blockDirectory))
+                    {
+                        blockDevices.put(blockDirectory, dataDirectory);
+                    }
+                }
+                catch (IOException e)
+                {
+                    logger.warn("IO exception while reading file {}.", dataDirectory, e);
+                }
+            }
+            for (Map.Entry<String, String> entry: blockDevices.entrySet())
+            {
+                String blockDeviceDirectory = entry.getKey();
+                String dataDirectory = entry.getValue();
+                try
+                {
+
+                    String readAheadKBPath = StartupChecks.getReadAheadKBPath(blockDeviceDirectory);
+
+                    final BufferedReader bufferedReader = Files.newBufferedReader(Path.of(readAheadKBPath));
```

Comment:
Perhaps we can have a slightly more succint check:
```
Path readAheadKBPath = StartupChecks.getReadAheadKBPath(blockDeviceDirectory);
if (readAheadKBPath != null && Files.exists(readAheadKBPath))
{
    int readAheadKbSetting = Ints.fromBytesArray(Files.readAllBytes(readAheadKBPath));
    // perform  check
}
```
wdyt?

Reply:
sure, refactored